auth4genai/snippets/get-started/langchain-next-js/call-others-api.mdx

```diff
@@ -1,479 +1,267 @@
-
-```tsx src/components/auth0-ai/TokenVault/TokenVaultInterruptHandler.tsx wrap lines
-import { TokenVaultInterrupt } from "@auth0/ai/interrupts";
-import type { Interrupt } from "@langchain/langgraph-sdk";
-
-import { TokenVaultConsent } from "@/components/auth0-ai/TokenVault";
-
-interface TokenVaultInterruptHandlerProps {
-  interrupt: Interrupt | undefined | null;
-  onFinish: () => void;
-}
-
-export function TokenVaultInterruptHandler({
-  interrupt,
-  onFinish,
-}: TokenVaultInterruptHandlerProps) {
-  if (
-    !interrupt ||
-    !TokenVaultInterrupt.isInterrupt(interrupt.value)
-  ) {
-    return null;
-  }
-
-  return (
-    <div key={interrupt.ns?.join("")} className="whitespace-pre-wrap">
-      <TokenVaultConsent
-        mode="popup"
-        interrupt={interrupt.value}
-        onFinish={onFinish}
-        connectWidget={{
-          title: "Authorization Required.",
-          description: interrupt.value.message,
-          action: { label: "Authorize" },
-        }}
-      />
-    </div>
-  );
-}
-```
-
-Now, update your chat window code to include the `TokenVaultInterruptHandler` component, for example:
-
-```tsx src/components/chat-window.tsx wrap lines highlight={2,3,8-19,36-38}
-//...
-import { TokenVaultInterruptHandler } from '@/components/auth0-ai/TokenVault/TokenVaultInterruptHandler';
-
-//... existing code
-export function ChatWindow(props: {
-  //... existing code
-}) {
-  const [threadId, setThreadId] = useQueryState('threadId');
-  const [input, setInput] = useState('');
-  const chat = useStream({
-    apiUrl: props.endpoint,
-    assistantId: 'agent',
-    threadId,
-    onThreadId: setThreadId,
-    onError: (e: any) => {
-      console.error('Error: ', e);
-      toast.error(`Error while processing your request`, { description: e.message });
-    },
-  });
-  //... existing code
-  return (
-    <StickToBottom>
-      <StickyToBottomContent
-        className="absolute inset-0"
-        contentClassName="py-8 px-2"
-        content={
-          chat.messages.length === 0 ? (
-            <div>{props.emptyStateComponent}</div>
-          ) : (
-            <>
-              <ChatMessages
-                aiEmoji={props.emoji}
-                messages={chat.messages}
-                emptyStateComponent={props.emptyStateComponent}
-              />
-              <div className="flex flex-col max-w-[768px] mx-auto pb-12 w-full">
-                <TokenVaultInterruptHandler interrupt={chat.interrupt} onFinish={() => chat.submit(null)} />
-              </div>
-            </>
-          )
-        }
-        {/* ... existing code */}
-      ></StickyToBottomContent>
-    </StickToBottom>
-  );
-}
-```
-Now when step-up authorization is required, the user will see a prompt in the chat window asking them to authorize.
+      ```tsx src/components/auth0-ai/FederatedConnections/FederatedConnectionInterruptHandler.tsx wrap lines
+      import { FederatedConnectionInterrupt } from "@auth0/ai/interrupts";
```

Comment:
Let's do a quick find for all existing `FederatedConnectionInterruptHandler` / `FederatedConnectionInterrupt` refs plz 🙏 - these are all adjusted after naming update